internal/transform/format.go

```diff
@@ -0,0 +1,492 @@
+}
+
+var pathOrder = []string{
+	"summary",
+	"description",
+	"servers",
+	"parameters",
+	"get",
+	"post",
+	"put",
+	"delete",
+	"patch",
+	"head",
+	"options",
+	"trace",
+}
+
+var operationOrder = []string{
+	"tags",
+	"summary",
+	"operationId",
+	"description",
+	"externalDocs",
+	"parameters",
+	"requestBody",
+	"responses",
+	"callbacks",
+	"deprecated",
+	"security",
+	"servers",
+}
+
+var schemaOrder = []string{
+	"name",
+	"type",
+	"title",
+	"summary",
+	"description",
+	"in",
+	"$ref",
+	"format",
+	"enum",
+	"default",
+	"multipleOf",
+	"maximum",
+	"exclusiveMaximum",
+	"minimum",
+	"exclusiveMinimum",
+	"maxLength",
+	"minLength",
+	"pattern",
+	"maxItems",
+	"minItems",
+	"uniqueItems",
+	"maxProperties",
+	"minProperties",
+	"required",
+	"properties",
+	"items",
+	"anyOf",
+	"oneOf",
+	"allOf",
+	"not",
+	"additionalProperties",
+	"example",
+	"examples",
+	"deprecated",
+}
+
+var responseOrder = []string{
+	"description",
+	"headers",
+	"content",
+	"links",
+}
+
+var securityOrder = []string{
+	"type",
+	"description",
+	"name",
+	"in",
+	"scheme",
+	"bearerFormat",
+	"flows",
+	"openIdConnectUrl",
+}
+
+var parameterOrder = []string{
+	"name",
+	"in",
+	"description",
+	"required",
+	"deprecated",
+	"allowEmptyValue",
+	"style",
+	"explode",
+	"allowReserved",
+	"schema",
+	"example",
+	"examples",
+	"content",
+}
+
+var requestBodyOrder = []string{
+	"description",
+	"content",
+	"required",
+}
+
+var headerOrder = []string{
+	"description",
+	"required",
+	"deprecated",
+	"allowEmptyValue",
+	"style",
+	"explode",
+	"schema",
+	"example",
+	"examples",
+	"content",
+}
+
+var tagOrder = []string{
+	"name",
+	"description",
+	"externalDocs",
+}
+
+var urlOrder = []string{
+	"name",
+	"identifier",
+	"url",
+	"description",
+	"variables",
+	"email",
+}
+
+var unknownOrder = []string{}
+
+// Orders contains the order of keys for each node type
+type Orders struct {
+	rootOrder        []string
+	infoOrder        []string
+	contactOrder     []string
+	componentsOrder  []string
+	pathOrder        []string
+	operationOrder   []string
+	schemaOrder      []string
+	responseOrder    []string
+	securityOrder    []string
+	parameterOrder   []string
+	requestBodyOrder []string
+	headerOrder      []string
+	tagOrder         []string
+	urlOrder         []string
+	unknownOrder     []string
+}
+
+var orders = Orders{
+	rootOrder:        rootOrder,
+	infoOrder:        infoOrder,
+	contactOrder:     contactOrder,
+	componentsOrder:  componentsOrder,
+	pathOrder:        pathOrder,
+	operationOrder:   operationOrder,
+	schemaOrder:      schemaOrder,
+	responseOrder:    responseOrder,
+	securityOrder:    securityOrder,
+	parameterOrder:   parameterOrder,
+	requestBodyOrder: requestBodyOrder,
+	headerOrder:      headerOrder,
+	tagOrder:         tagOrder,
+	urlOrder:         urlOrder,
+	unknownOrder:     unknownOrder,
+}
+
+// NodeTypeIdentifier contains sets of keys that identify different node types
+var NodeTypeIdentifier = struct {
```

Comment:
Nice

Reply:
This section was quite annoying, I never realized how much overlap there was in keys for different levels of the spec, so these arrays as well as the order of detection in the switch changed many times.